base/compiler/tfuncs.jl

```diff
@@ -2488,6 +2488,12 @@ function builtin_effects(𝕃::AbstractLattice, @nospecialize(f::Builtin), argty
         return Effects(EFFECTS_TOTAL;
             consistent = (isa(setting, Const) && setting.val === :conditional) ? ALWAYS_TRUE : ALWAYS_FALSE,
             nothrow = compilerbarrier_nothrow(setting, nothing))
+    elseif f === Core.current_scope
+        length(argtypes) == 0 || return Effects(EFFECTS_THROWS; consistent=ALWAYS_FALSE)
```

Comment:
`argtypes == Any[Vararg]` is valid here too

Reply:
And if `length(argtypes) >= 1 && !isvarargtype(argtypes[1])`, then isn't it just `EFFECTS_THROWS`?